src/function/gds/all_shortest_paths.cpp

```diff
@@ -18,15 +18,15 @@ namespace kuzu {
 namespace function {
 
 class ParentList {
```

Comment:
Not in this PR but I think we should think about how we should implement the optimization when edgeIDs are not needed. In that case we should not keep track of them here or in the FactorizedTables. What would be a simple way to implement that optimization? That can decrease the memory footprint of these algorithms by more than 1/3 and should just speed things up. Would be good to think about this now and put that optimization and your general approach in this [issue](https://github.com/kuzudb/kuzu/issues/4285).

Reply:
Left a TODO in the codebase.